src/main/scala/higherkindness/rules_scala/workers/common/AnnexScalaInstance.scala

```diff
@@ -67,11 +67,20 @@ object AnnexScalaInstance {
    * any different from leaving Zinc's cache enabled.
    */
   private def getAnnexScalaInstance(allJars: Array[File], workDir: Path): AnnexScalaInstance = {
-    // We need to remove the sandbox prefix from the paths in order to compare them.
-    val mapBuilder = Map.newBuilder[Path, Path]
+    // We want to compare short paths to avoid the Bazel sandbox prefix and arch/config specific parts of the path
+    // We use a tree map because we want the keys sorted for comparison purposes
+    val mapBuilder = TreeMap.newBuilder[Path, Path]
```

Comment:
I use the map later on in this file as I need to copy the jar from its original location to its destination location.

I use the string for the keys because classpaths typically are jars concatenated with `:`.

I have to loop through the keys once anyway to make the string. I have to loop through the keys once to make the set.

Benefit of the set is that I don't have to insert things into a treemap, so that should be faster.

Overall I think using a map and the set as the cache key will be faster and safer. I'll make some updates.

Reply:
Or even better I can make the key while making the map provided set has a builder. That would be a nice micro optimization.

Reply:
True, that would be a good optimization. Even if constructing a `Set` turns out to be slightly slower because we can't reuse the result of `map`ing through, I think the safety benefits outweigh the performance detriments.

Reply:
I ended up using an unsorted map and a set. I use the set for the key and that seems to work great. I also tried using the `.values` Iterable as the key from the sorted map, but that resulted in cache misses. Didn't bother to dig in and find out why. I figure creating a set and an unsorted map is probably roughly the same cost as creating a sorted map.